File: hw/tipan/u3/hw25CompositePattern/CompositePatternPython/controller/employee_controller.py
from model.clerk import Clerk
from model.teller import Teller
from model.manager import Manager
from model.president import President


class EmployeeController:

    def build_organization(self):

        lonny = Teller("Lonny")
        cal = Clerk("Cal")
        susan = Teller("Susan")
        david = Clerk("David")

        able = Manager("Able")
        able.add(lonny)
        able.add(cal)
        able.add(susan)
        able.add(david)

        juanita = Teller("Juanita")
        tina = Teller("Tina")
        thelma = Teller("Thelma")
        kevin = Clerk("Kevin")
        laura = Clerk("Laura")

        becky = Manager("Becky")
        becky.add(juanita)
        becky.add(tina)
        becky.add(thelma)
        becky.add(kevin)
        becky.add(laura)

        robert = Teller("Robert")
        emily = Teller("Emily")
        sophia = Clerk("Sophia")
        daniel = Clerk("Daniel")

        michael = Manager("Michael")
        michael.add(robert)
        michael.add(emily)
        michael.add(sophia)
        michael.add(daniel)

        pete = President("Pete")
        pete.add(able)
        pete.add(becky)
        pete.add(michael)

        return pete

File: hw/tipan/u3/hw25CompositePattern/CompositePatternJavaScript/controller/EmployeeController.js
const Clerk = require("../model/Clerk");
const Teller = require("../model/Teller");
const Manager = require("../model/Manager");
const President = require("../model/President");

class EmployeeController {

    buildOrganization() {

        const lonny = new Teller("Lonny");
        const cal = new Clerk("Cal");
        const susan = new Teller("Susan");
        const david = new Clerk("David");

        const able = new Manager("Able");
        able.add(lonny);
        able.add(cal);
        able.add(susan);
        able.add(david);

        const juanita = new Teller("Juanita");
        const tina = new Teller("Tina");
        const thelma = new Teller("Thelma");
        const kevin = new Clerk("Kevin");
        const laura = new Clerk("Laura");

        const becky = new Manager("Becky");
        becky.add(juanita);
        becky.add(tina);
        becky.add(thelma);
        becky.add(kevin);
        becky.add(laura);

        const robert = new Teller("Robert");
        const emily = new Teller("Emily");
        const sophia = new Clerk("Sophia");
        const daniel = new Clerk("Daniel");

        const michael = new Manager("Michael");
        michael.add(robert);
        michael.add(emily);
        michael.add(sophia);
        michael.add(daniel);

        const pete = new President("Pete");
        pete.add(able);
        pete.add(becky);
        pete.add(michael);

        return pete;
    }

}

module.exports = EmployeeController;